### PR TITLE
Sql server migration for local development using Sql server inside docker container

### LINKED ISSUE #130: As a developer I would like to use SQL server database for the locally hosted program instantiated via docker
Acceptance:
- The SQL database for local development is now using SQL server as the database component

- [ ] - Set up Docker with the sql server database
- [ ] - Tests still run successfully after the change
- [ ] - Ensure migrations are being used when deploying to app services

### PR DESCRIPTION
**Pull request by Slug-Boi**
This implementation of SQL server with docker containers takes care of the local development side of things. This more than likely breaks the tests which will have to be fixed on a separate branch which is another issue #134 
Resolves #130

### CODE REVIEW

**Review by August-Brandt — APPROVED** [review]
It all looks good. Nice work

**Review by JonasSkjodt — APPROVED** [review]
looks good. The tests, as you say, can be handled in another branch.

**Review by Alex-itu — APPROVED** [review]

File: src/Chirp.Web/appsettings.json
Comment (Haruyzal) [inline src/Chirp.Web/appsettings.json:1]: I don't think I understand why "ConnectionStrings" is empty? :)

**Review by Haruyzal — CHANGES_REQUESTED** [review]
It seems errors occur 'cause of unremoved usings in our tests, where the .csproj doesn't recognise these anymore:
- `InMemoryWebApplicationRactory.cs` -> `Using Microsoft.Data.Sqlite`
- `ChirpRazorHttpClient_UnitTests.cs` -> `Using SimpleDB`

Banger work 🔥

File: src/Chirp.Web/appsettings.json
Comment (Slug-Boi) [reply src/Chirp.Web/appsettings.json:1]: Cause we are now using user-secrets forgot to say that in the PR

> It seems errors occur 'cause of unremoved usings in our tests, where the .csproj doesn't recognise these anymore:...

Most of the test need to be rewritten from the ground up to support SQL Server so think we're just going to wait with changing test stuff for another branch

Comment (Slug-Boi) [reply src/Chirp.Web/appsettings.json:1]: Or are those errors happening with Dotnet run? If so just remove them and push it to the branch =)

**Comment by Slug-Boi** [comment]
I have found a pretty big problem. When changing over to SQL server the Async methods used to access and change the DBContext now run into concurrency problems. Resulting in an exception being thrown when trying to access an authors timeline

File: src/Chirp.Web/appsettings.json
Comment (Haruyzal) [reply src/Chirp.Web/appsettings.json:1]: > Cause we are now using user-secrets forgot to say that in the PR
But when I run `dotnet user-secrets list --project src/Chirp.Web`, it replies: `No secrets configured for this application.` Effectively making the ConnectionStrings empty, at least on my end ^^"

Comment (Slug-Boi) [reply src/Chirp.Web/appsettings.json:1]: I'll give a small presentation today on how to do it but you need to individually set the Conn string and then use docker to run the server which will allow you to get said Conn string

**Review by Haruyzal — APPROVED** [review]
New changes look good, awesome work guys.
- Approval sent from my iPhone.

**Pull request merged**